lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -2305,6 +2313,12 @@ balanceTransaction ctx genChange (ApiT wid) body = do
     txIncoming = decodeTx tl sealedTxIncoming
     toTxOut (TxChange addr amt assets _) =
         TxOut addr (TokenBundle amt assets)
+    getExternalInps (ApiExternalInput (ApiTxIn (ApiT hashTx) ix) (ApiTxOut _ _ (Quantity amt) _)) =
+        (TxIn hashTx ix, Coin $ fromIntegral amt)
+    externalIns = getExternalInps <$> apiExternalInps
+    areOutputsCovered (Tx _ feeM _ _ outs _ _ _) execUnitsCost =
+        sumCoins (map snd $ L.nubBy (\x y -> fst x == fst y) externalIns) ==
```

Comment:
maybe we need here `externalIns ++ colls ++ inps` the last two from `(Tx _ feeM colls inps outs _ _ _)` :thinking:  but then the unpended example does not work (and this is just complete normal tx)